src/bin/genemichaels.rs

```diff
@@ -199,7 +237,104 @@ fn main() {
             Offable::On(_) => true,
         },
     };
-    if args.files.is_empty() {
+    if args.workspace || args.folder {
+        // can use the same code for both
+        let res = || -> Result<()> {
+            {
+                let inst = time::Instant::now();
+                eprintln!("\x1B[1;32m  Formatting\x1B[0;22m {}...", if args.folder {
+                    "folder"
+                } else {
+                    "workspace"
+                });
+                let mut workspace_cargo_toml = None;
+                let c_dir = current_dir()?;
+                if args.workspace {
+                    let mut at: Option<&Path> = Some(&c_dir);
+                    while let Some(d) = at.take() {
+                        if d.join("Cargo.toml").exists() &&
```

Comment:
Mind putting the path in a variable? Just to avoid accidental typos from having it written out multiple times.

Reply:
Good call, put in a constant. :+1: